File: Data Structures & Algorithms/climbing-stairs/submission-2.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 2:
            return 1

        x0 = 0
        x1 = 1
        x2 = 1

        i = 1
        while i <= n:
            x2 = x0 + x1
            x0 = x1
            x1 = x2
            i += 1
        
        return x2

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k)
    {
        unordered_map<int, int> numCount;
        for (const auto &n : nums)
        {
            if (numCount.contains(n))
            {
                numCount[n]++;
            }
            else
            {
                numCount[n] = 1;
            }
        }

        std::vector<std::pair<int, int>> numCountVec;
        numCountVec.reserve(numCount.size());

        for(const auto &[key, value] : numCount)
        {
            numCountVec.push_back(std::pair<int, int>(key, value));
        }

        sort(numCountVec.begin(), numCountVec.end(), 
            [](std::pair<int, int> kv1, std::pair<int, int> kv2)
            {
                return kv1.second > kv2.second;
            }
        );

        vector<int> output(k);
        for(int i=0; i<k; ++i)
        {
            output[i] = numCountVec[i].first;
        }

        return output;
    }
};
